### ISSUE TITLE
Add operators support for `$addFields`

### ISSUE THREAD

**Issue by noisersup** [FerretDB/FerretDB#2679]
After #2678.

### What should be done?

Add operators support to `$addFields`.

#### Supported stages support operators
- `$group` stage `_id` expression supports operators
example: `{_id : { $dateToString: { format: "%Y-%m-%d", date: "$date" } } }` from [manual](https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/#group-by-day-of-the-year)
https://github.com/FerretDB/FerretDB/blob/0992e5f8a410e59041a62a12f666eacf6f3bd27b/internal/handlers/common/aggregations/stages/group.go#L165 
- `$project` stage supports operators
example: `{$project: {isbn: {prefix: { $substr: [ "$isbn", 0, 3 ] } } } }` from [manual](https://www.mongodb.com/docs/manual/reference/operator/aggregation/project/#include-computed-fields)
https://github.com/FerretDB/FerretDB/blob/0992e5f8a410e59041a62a12f666eacf6f3bd27b/internal/handlers/common/aggregations/stages/projection/projection.go#L361
- query projection supports [operators](https://www.mongodb.com/docs/manual/reference/method/db.collection.find/#projection)
same example as `$project` works on projection
https://github.com/FerretDB/FerretDB/blob/0992e5f8a410e59041a62a12f666eacf6f3bd27b/internal/handlers/common/projection.go#L371

#### Out of scope
- `$match` is handled by https://github.com/FerretDB/FerretDB/issues/414
- `$group` stage recursive operator is handled by https://github.com/FerretDB/FerretDB/issues/2694

### Definition of Done

- all stages updated;
- compatibility tests for every operator in every stage added;
- spot refactorings done;
- find issue number in the code, skipped tests with this issue number passes.

**Comment by noisersup** [comment]
Yeah I wanted to describe that but I did the opposite, nice catch :)
